litmus-portal/frontend/src/pages/MonitoringDashboardPage/index.tsx

```diff
@@ -70,19 +145,129 @@ const DashboardPage: React.FC = () => {
     dashboardListForAgent: [],
     metaData: [],
     dashboardKey: 'Default',
+    selectionOverride: false,
+  });
+  const [
+    prometheusQueryData,
+    setPrometheusQueryData,
+  ] = React.useState<PrometheusQueryDataInterface>({
+    promInput: {
+      url: '',
+      start: '',
+      end: '',
+      queries: [],
+    },
+    chaosInput: [],
+    chaosEvents: [],
+    chaosEventsToBeShown: [],
+    numOfWorkflows: 0,
+    firstLoad: true,
   });
+  const [eventsToShow, setEventsToShow] = React.useState<EventsToShowInterface>(
+    {
+      eventsToShow: [],
+      selectEvents: false,
+    }
+  );
   const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
-  const [refreshRate, setRefreshRate] = React.useState<number>(10000);
+  const [refreshRate, setRefreshRate] = React.useState<number>(
+    selectedDashboard.refreshRate && !prometheusQueryData.firstLoad
+      ? selectedDashboard.refreshRate
+      : 0
+  );
   const [dataSourceStatus, setDataSourceStatus] = React.useState<string>(
     'ACTIVE'
   );
   const open = Boolean(anchorEl);
+  const [chaosDataSet, setChaosDataSet] = React.useState<ChaosDataSet>({
+    queryIDs: [],
+    chaosData: [],
+    visibleChaos: [],
+    latestEventResult: [],
+  });
   const handleClick = (event: React.MouseEvent<HTMLElement>) => {
     setAnchorEl(event.currentTarget);
   };
   const handleClose = () => {
     setAnchorEl(null);
   };
+  const dateRangeSelectorRef = React.useRef<HTMLButtonElement>(null);
+  const [
+    isDateRangeSelectorPopoverOpen,
+    setDateRangeSelectorPopoverOpen,
+  ] = React.useState(false);
+
+  const clearTimeOuts = async () => {
+    let id = window.setTimeout(() => {}, 0);
+    while (id--) {
+      window.clearTimeout(id);
+    }
+
+    return Promise.resolve(id === 0);
+  };
+
+  const CallbackFromRangeSelector = (
+    selectedStartDate: string,
+    selectedEndDate: string
+  ) => {
+    const startDateFormatted: string = moment(selectedStartDate).format();
+    const endDateFormatted: string = moment(selectedEndDate)
+      .add(23, 'hours')
+      .add(59, 'minutes')
+      .add(59, 'seconds')
```

Comment:
Can `hours minutes, seconds `  be taken from translation? If moment allows it then please check it

Reply:
Moment only allows english in function parameters.